deploy/transport-controller-deployment.yaml.template

```diff
@@ -0,0 +1,90 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: transport-controller-config
+  namespace: ${WDS_NAME}-system
+data:
+  get-kubeconfig.sh: |
+    #!/bin/bash
+    # this script receives a ControlPlane name and a parameter
+    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
+    # or the external kubeconfig (if set to "true", the first will be retrieved).
+    # The function returns the requested kubeconfig's content in base64.
+    # it assumes the kubeconfig context is set to access the hosting cluster.
+
+    cpname="$1"
+    get_incluster_key="$2"
+
+    key=""
+    if [[ $get_incluster_key == "true" ]]; then
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
+    else
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
+    fi
+
+    # get secret details
+    secret_name=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.name}')
+    secret_namespace=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.namespace}')
+
+    # get the kubeconfig in base64
+    kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}"
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: transport-controller
+  namespace: ${WDS_NAME}-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: transport-controller
+  template:
+    metadata:
+      labels:
+        name: transport-controller
+    spec:
+      serviceAccountName: kubestellar-controller-manager
+      initContainers:
```

Comment:
Maybe I'm missing something. Why do we need two different init containers and can't generate both files using a single one ?

Reply:
it's possible.
I did it just for separation of concerns, for example - If I want to run IMBS in kind cluster (just ocm hub in kind) I can easily remove the imbs-setup init container and mount imbs kubeconfig in a secret instead.
it can be done in a single init container but I like more the separation since the way we get WDS kubeconfig is not necessarily identical to how we get the IMBS kubeconfig.

Reply:
@ezrasilvera not if two different setups require using different images (e.g, image that contains some tool). 
if there is no issue with keeping two separate init containers, I prefer keeping the two

Reply:
Having the init in two containers does not bother me.